gamemodes/terrortown/gamemode/server/sv_roleselection.lua

```diff
@@ -460,18 +460,19 @@ function roleselection.SelectRoles(plys, maxPlys)
 	local selectedForcedPlys = SelectForcedRoles(plys, selectableRoles) -- this updates roleselection.finalRoles table and returns a key based list of selected players
 
 	-- We need to remove already selected players
-	local tmpPlys = {} -- modified player table
+	local plysFirstPass, plysSecondPass = {}, {} -- modified player table
 
 	for i = 1, #plys do
 		local ply = plys[i]
 
 		if not selectedForcedPlys[ply] then
-			tmpPlys[#tmpPlys + 1] = ply
+			plysFirstPass[#plysFirstPass + 1] = ply
```

Comment:
Bc `plysFirstPass` is the same as `plysSecondPass`, you should localize `#plysFirstPass + 1`.